src/components/HeaderNavMenu.test.jsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import HeaderNavMenu from './HeaderNavMenu';
+import { BrowserRouter as Router } from 'react-router-dom';
+
+const links = {
+  topology: '/topology',
+  'abstract-algebra': '/abstract-algebra',
+  'differential-geometry': '/differential-geometry',
+};
+
+describe('HeaderNavMenu', () => {
+  test('renders App component', () => {
+    render(
+      <Router>
+        <HeaderNavMenu links={links} />
+      </Router>
+    );
+  });
+  test('renders Header links', () => {
+    render(
+      <Router>
+        <HeaderNavMenu links={links} />
+      </Router>
+    );
+    expect(screen.getByText('topology').closest('a')).toHaveAttribute(
+      'href',
+      '/topology'
+    );
```

Comment:
You can use `getByRole` and do something like

`expect(
      screen.getByRole('tab', { name: 'topology' }).getAttribute('href')
    ).toBe('/topology');`

Reply:
what's wrong with what I had?